draft/elm.md

```diff
@@ -0,0 +1,30 @@
+
+In 2015 functional (reactive) programming advanced in the frontend field and by now every frontend developer should have heard of it. Functional programming paradigms and ideas are influencing JavaScript and its frameworks: [PureScript](http://www.purescript.org/) and [RxJS](https://github.com/Reactive-Extensions/RxJS) are examples for languages and libraries that build on the functional-reactive concepts – [React](https://facebook.github.io/react/) and accompanying libraries are increasingly influenced by the functional trend as well.
+
+But instead of mounting a framework for working in a functional manner on top of our good old JavaScript, Elm is a whole new language that builds on functional paradigms and makes reactive programming a core concept. And even so it compiles to JavaScript so it can run in the browser, Elm does not try to bring functional programming to JavaScript, but to its developers.
+
+### Statically typed, no more runtime errors.
+
+Elm is a functional programming language that compiles to JavaScript. Opposed to JavaScript, Elm is statically typed. The compiler type checks the code and guarantees to generate reliable code that will not throw any runtime exceptions caused by inconsisted types or undefined variables. The compiler is the centerpiece of Elm and enables many of the interesting features like [very helpful error messages](http://elm-lang.org/blog/compiler-errors-for-humans) that also contain hints for fixing the code or a [package manager that enforces semver](http://elm-lang.org/blog/announce/package-manager) through code analysis.
+
+Consequently even the view is statically typed – the templates, which are written in elm-html, are based on the concept of the Virtual DOM, just like in React. Regarding the rendering, elm-html [performs even better than React](http://elm-lang.org/blog/blazing-fast-html) and other JavaScript frameworks, which is due to data immutability and ruling out side effects by usage of pure functions. But rather than comparing the performance the main advantage has to be seen in the fact that [the Elm architecture](https://github.com/evancz/elm-architecture-tutorial) suggests and encourages encapsulation of the view functionality into modular components, which makes them combinable and leads to the creating of reusable modules.
+
+The Elm syntax might look odd at first if you have not worked with a functional programming language before – nevertheless it is very clear and does not take much time to get used to in my opinion. Elms standard library is relatively small compared to other languages, but rather than the syntax and core modules its the concepts which need to be learned and understood. Amongst others these are the Elm architecture, immutability and purity as well as [modelling problems the Elm way](http://elm-lang.org/guide/model-the-problem).
```

Comment:
Maybe a snippet of code would do well to at least show what it looks like before starting this paragraph.